test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -842,6 +714,142 @@ define([
+                        [jcm.PARAM.JOB_ID_LIST]: reqData.allIds,
+                    };
+                    return {
+                        type: `${type} error, ${req} params`,
+                        message: errorMessage,
+                        expected: reqData.channelsNoMap.map((channel) => {
+                            return [
+                                errorMessage,
+                                {
+                                    key: { type: jcm.MESSAGE_TYPE.ERROR },
+                                    channel,
+                                },
+                            ];
+                        }),
+                        expectedWithMap: reqData.channelsNoMap.map((channel) => {
+                            return [
+                                errorMessage,
+                                {
+                                    key: { type: jcm.MESSAGE_TYPE.ERROR },
+                                    channel,
+                                },
+                            ];
+                        }),
+                    };
+                }
+
+                const busTests = [];
+                ['simple', 'complex'].forEach((type) => {
+                    Object.keys(requests).forEach((req) => {
+                        busTests.push(generateErrorMessage(type, req));
+                    });
+                });
+                // these messages do not get passed on to the frontend as they're not very useful
+                Object.keys(requests).forEach((req) => {
+                    const test = generateErrorMessage('jobRequestException', req);
+                    test.expected = [];
+                    test.expectedWithMap = [];
+                    busTests.push(test);
+                });
+
+                // no mapping in place
+                busTests.forEach((test) => {
+                    it(`should send a ${test.type} message to the bus`, () => {
+                        const msg = makeCommMsg(jcm.MESSAGE_TYPE.ERROR, test.message),
+                            comm = new JobCommChannel();
+                        return comm.initCommChannel().then(() => {
+                            spyOn(testBus, 'send');
+                            comm.handleCommMessages(msg);
+                            expect(testBus.send.calls.allArgs()).toEqual(
+                                jasmine.arrayWithExactContents(test.expected)
+                            );
+                            // create the mapping and handle the message again
+                            comm._jobMapping = jobMapping;
+                            comm.handleCommMessages(msg);
+                            expect(testBus.send.calls.allArgs()).toEqual(
+                                jasmine.arrayWithExactContents(
+                                    test.expected.concat(test.expectedWithMap)
+                                )
+                            );
```

Comment:
test each message as if (1) there was no job mapping in place, and (2) as if a mapping had been created by previous jobs